src/components/Quiz.js

```diff
@@ -1,56 +1,112 @@
-import React, { useEffect, useState } from "react";
+import React, { useState } from "react";
+import Typography from '@mui/material/Typography';
+import ImageButton from './ImageButton'; 
+import ImageOverlay from './ImageOverlay'; 
 
 const Quiz = ({ questions, weights, updateWeights, setIsQuizComplete }) => {
   const [currentQuestionIndex, setCurrentQuestionIndex] = useState(0);
-  const [currentQuestion, setCurrentQuestion] = useState(null);
-
-  useEffect(() => {
-    setCurrentQuestion(questions[currentQuestionIndex]);
-  }, [currentQuestionIndex, questions]);
+  const [selectedChoices, setSelectedChoices] = useState({}); 
+  const [answeredQuestions, setAnsweredQuestions] = useState(new Array(questions.length).fill(false)); 
 
   const handleChoiceSelection = (selectedChoice) => {
+    const previousChoice = selectedChoices[currentQuestionIndex];
     const updatedWeights = { ...weights };
 
+    if (previousChoice) {
+      for (const [key, value] of Object.entries(previousChoice.weights)) {
+        const normalizedKey = key.toLowerCase();
+        updatedWeights[normalizedKey] =
+          (updatedWeights[normalizedKey] || 0) - value;
+      }
+    }
+
     for (const [key, value] of Object.entries(selectedChoice.weights)) {
       const normalizedKey = key.toLowerCase();
       updatedWeights[normalizedKey] =
         (updatedWeights[normalizedKey] || 0) + value;
     }
 
     updateWeights(updatedWeights);
+    setSelectedChoices({
+      ...selectedChoices,
+      [currentQuestionIndex]: selectedChoice,
+    });
+    setAnsweredQuestions(answeredQuestions.map((answered, index) => index === currentQuestionIndex ? true : answered));
+  };
 
-    const nextQuestionIndex = currentQuestionIndex + 1;
-    if (nextQuestionIndex < questions.length) {
-      setCurrentQuestionIndex(nextQuestionIndex);
-    } else {
+  const handleStepClick = (index) => {
```

Comment:
Unless you plan on doing something more with this down the road, this function feels unnecessary since all it does is call another function with the same variable being passed into it.

Reply:
that code will be used in the next PRs where I added a progress navbar to my quiz questions found in this branch: https://github.com/Miramoop/Quiz-Tunes-React/tree/feature/addition-of-quiz-question-navbar